src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -37,10 +37,31 @@ public static bool RequiresReflectionMethodBodyScannerForCallSite (LinkContext c
 			if (methodDefinition == null)
 				return false;
 
-			return
-				GetIntrinsicIdForMethod (methodDefinition) > IntrinsicId.RequiresReflectionBodyScanner_Sentinel ||
+			bool isPInvokeImpl = methodDefinition.IsPInvokeImpl;
+			if (isPInvokeImpl) {
+				// HACK - we don't want to scan C++/CLI assemblies
+				if (IsCPPCLIAssembly (methodDefinition.DeclaringType.Module)) {
+					isPInvokeImpl = false;
+				}
+				// @TODO - there must be a better way to find out methods from C++/CLI assemblies
+				// Revisit after discussing with the team
+				static bool IsCPPCLIAssembly (ModuleDefinition module)
+				{
+					foreach (var type in module.Types)
+						if (type.Namespace == "<CppImplementationDetails>" ||
+							type.Namespace == "<CrtImplementationDetails>")
+							return true;
+
+					return false;
+				}
+			}
+
+			bool requiredCheck = GetIntrinsicIdForMethod (methodDefinition) > IntrinsicId.RequiresReflectionBodyScanner_Sentinel ||
 				context.Annotations.FlowAnnotations.RequiresDataFlowAnalysis (methodDefinition) ||
```

Comment:
If the PInvoke check is expensive, consider adding it into the FlowAnnotations - that has a cache already built-in. We would only need to pay the price once per method then.